pkg/querier/worker/scheduler_processor.go

```diff
@@ -114,23 +125,31 @@ func (sp *schedulerProcessor) querierLoop(c schedulerpb.SchedulerForQuerier_Quer
 			return err
 		}
 
+		inflightQuery.Store(true)
+
 		// Handle the request on a "background" goroutine, so we go back to
 		// blocking on c.Recv().  This allows us to detect the stream closing
 		// and cancel the query.  We don't actually handle queries in parallel
 		// here, as we're running in lock step with the server - each Recv is
 		// paired with a Send.
 		go func() {
+			defer inflightQuery.Store(false)
+
 			// We need to inject user into context for sending response back.
-			var (
-				ctx    = user.InjectOrgID(ctx, request.UserID)
-				logger = util_log.WithContext(ctx, sp.log)
-			)
+			ctx := user.InjectOrgID(ctx, request.UserID)
 
-			sp.metrics.inflightRequests.Inc()
+			tracer := opentracing.GlobalTracer()
+			// Ignore errors here. If we cannot get parent span, we just don't create new one.
+			parentSpanContext, _ := httpgrpcutil.GetParentSpanForRequest(tracer, request.HttpRequest)
+			if parentSpanContext != nil {
+				queueSpan, spanCtx := opentracing.StartSpanFromContextWithTracer(ctx, tracer, "querier_processor_runRequest", opentracing.ChildOf(parentSpanContext))
+				defer queueSpan.Finish()
 
-			sp.runRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, request.HttpRequest)
+				ctx = spanCtx
+			}
+			logger := util_log.WithContext(ctx, sp.log)
 
-			sp.metrics.inflightRequests.Dec()
```

Comment:
Where did this go?

Reply:
Hi, this was my mistake, I simply ‘copied’ scheduler mimir's code to loki, ignoring the inflightRequests metrics. 
I have submitted a commit which fixes this issue.